de4a-connector/src/main/java/eu/de4a/connector/as4/client/DomibusGatewayClient.java

```diff
@@ -51,179 +56,185 @@
+        partInfo2.setHref(idCanonical);
+
+        Property prop = new Property();
+        prop.setName("MimeType");
+        prop.setValue(MediaType.APPLICATION_XML_VALUE);
+
+        PartProperties props = new PartProperties();
+        props.getProperty().add(prop);
+        partInfo.setPartProperties(props);
+        partInfo2.setPartProperties(props);
+        attachments.add(partInfo);
+        attachments.add(partInfo2);
+
+        return attachments;
+    }
+
+    private MessageOwner buildMessageOwner(LargePayloadType payload, ResponseAndHeader response) {
+        byte[] targetArray = null;
+        try {
+            targetArray = IOUtils.toByteArray(payload.getValue().getDataSource().getInputStream());
+            if (targetArray != null && Base64.isBase64(targetArray)) {
+                targetArray = Base64.decodeBase64(targetArray);
+            }
+
+            Document doc = DOMUtils.byteToDocument(targetArray);
+            MessageOwner messageOwner = new MessageOwner();
+            messageOwner.setMessage(doc.getDocumentElement());
+            String idrequest = null;
+            idrequest = DOMUtils.getValueFromXpath(DE4AConstants.XPATH_ID, doc.getDocumentElement());
+            if (ObjectUtils.isEmpty(idrequest)) {
+                LOGGER.error("EvidenceRequest without requestId");
+            } else {
+                messageOwner.setId(idrequest);
+                SimpleParticipantIdentifier receiver = SimpleIdentifierFactory.INSTANCE.createParticipantIdentifier(
+                        response.getInfo().getUserMessage().getPartyInfo().getTo().getPartyId().getType(), 
+                        response.getInfo().getUserMessage().getPartyInfo().getTo().getPartyId().getValue());
+                messageOwner.setReceiverId(receiver.getURIEncoded());
+                SimpleParticipantIdentifier sender = SimpleIdentifierFactory.INSTANCE.createParticipantIdentifier(
+                        response.getInfo().getUserMessage().getPartyInfo().getFrom().getPartyId().getType(), 
+                        response.getInfo().getUserMessage().getPartyInfo().getFrom().getPartyId().getValue());
+                messageOwner.setSenderId(sender.getURIEncoded());
+                return messageOwner;
+            }
+        } catch (MessageException | IOException e) {
+            LOGGER.error("Error retrieving bytes from domibus response", e);
+        }
+        return null;
+    }
+    
+    private void processMessage(String requestId) {
+        ResponseAndHeader response = domibusClientWS.getMessageWithHeader(requestId, domibusEndpoint);
+        LargePayloadType payload = null;
+        if (response != null && response.getInfo() != null) {
+            RetrieveMessageResponse message = response.getMessage();
+
+            payload = message.getPayload().stream()
+                    .filter(p -> p.getPayloadId().contains(DE4AConstants.TAG_EVIDENCE_REQUEST)).findFirst()
+                    .orElse(null);
```

Comment:
This could be replaced with an .ifPresentOrElse(() -> { action if present }, () -> { action if not present });
Also a bit more concise when using streams and sort of goes full circle...